File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.java
import java.util.*;
class Solution {
    
    public int solution(int[] ingredient) {
        int ret = 0;
        String s = "1231";
        int start = 0;
        int end = start + 3;
        Stack<Integer> stk = new Stack<>();
        for(int i : ingredient){
            stk.push(i);
            if(stk.size() >= 4){
                String temp = "";
                while(!stk.isEmpty() && temp.length() < 4) temp = stk.pop() + temp;
                if(temp.equals(s)) ret++;
                else {
                    for(char c : temp.toCharArray()) stk.push(c - '0');
                }
            }
        }
        
        return ret;
    }
}

File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.py
def solution(ingredient):
    ret = 0
    stk = []
    for c in ingredient:
        stk.append(str(c))
        if len(stk) >= 4 and stk[-1] == '1':
            temp = ''
            for _ in range(4):
                temp = stk.pop() + temp
            if temp == '1231':
                ret += 1     
                continue
            for c in temp:
                stk.append(c)
    return ret
    
                
                
                
            

"""
1: 빵
2: 야채
3: 고기
1231

"""